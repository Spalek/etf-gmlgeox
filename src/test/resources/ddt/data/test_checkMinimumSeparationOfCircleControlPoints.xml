<ii:GeometryCollection xmlns:gml="http://www.opengis.net/gml/3.2"
 xmlns:ii="http://www.interactive-instruments.de/test" xmlns:xlink="http://www.w3.org/1999/xlink"
 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 <gml:boundedBy>
  <gml:Envelope srsName="http://www.opengis.net/def/crs/EPSG/0/4326">
   <gml:lowerCorner>0 0</gml:lowerCorner>
   <gml:upperCorner>90 90</gml:upperCorner>
  </gml:Envelope>
 </gml:boundedBy>
 <ii:member>
  <gml:Surface gml:id="S1">
   <gml:patches>
    <gml:PolygonPatch>
     <gml:exterior>
      <gml:Ring>
       <gml:curveMember>
        <gml:Curve>
         <gml:segments>
          <gml:Circle>
           <gml:posList>-1 0 0 -1 0 1</gml:posList>
          </gml:Circle>
         </gml:segments>
        </gml:Curve>
       </gml:curveMember>
      </gml:Ring>
     </gml:exterior>
    </gml:PolygonPatch>
   </gml:patches>
  </gml:Surface>
 </ii:member>
 <ii:member>
  <gml:Surface gml:id="S2">
   <gml:patches>
    <gml:PolygonPatch>
     <gml:exterior>
      <gml:Ring>
       <gml:curveMember>
        <gml:Curve>
         <gml:segments>
          <gml:Circle>
           <gml:posList>-10 0 0 1 10 0</gml:posList>
          </gml:Circle>
         </gml:segments>
        </gml:Curve>
       </gml:curveMember>
      </gml:Ring>
     </gml:exterior>
    </gml:PolygonPatch>
   </gml:patches>
  </gml:Surface>
 </ii:member>
</ii:GeometryCollection>
