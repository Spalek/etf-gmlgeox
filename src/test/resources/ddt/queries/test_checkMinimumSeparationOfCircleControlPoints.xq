import module namespace ggeo = 'https://modules.etf-validator.net/gmlgeox/2';

declare namespace gml = 'http://www.opengis.net/gml/3.2';
declare namespace ii = 'http://www.interactive-instruments.de/test';

declare variable $dbname external := 'GMLGEOX-JUNIT-TEST-DB-000';
let $members := db:open($dbname)//ii:member
let $init := ggeo:init($dbname)

let $geometries := $members/*[.//*[self::gml:Circle]]
return
 <test_checkMinimumSeparationOfCircleControlPoints>    
   {
    for $geom in $geometries
    let $arcStrings := $geom//*[self::gml:Circle]
    let $invalidCase :=
     (for $arcString in $arcStrings
      return
        ggeo:checkMinimumSeparationOfCircleControlPoints($arcString,60)
      )[position() le 1]
    let $result := 
      if($invalidCase) then
        'failed'
      else
        'passed'
    return 
      <test geom='{$geom/@gml:id}'>{$result}</test>    
   }
 </test_checkMinimumSeparationOfCircleControlPoints>
