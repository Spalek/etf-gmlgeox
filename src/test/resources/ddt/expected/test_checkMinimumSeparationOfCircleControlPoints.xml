<test_checkMinimumSeparationOfCircleControlPoints>
  <test geom="S1">passed</test>
  <test geom="S2">failed</test>
</test_checkMinimumSeparationOfCircleControlPoints>